src/main/java/it/gov/pagopa/pu/debtpositions/service/delete/DebtPositionDeletionServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+import it.gov.pagopa.pu.debtpositions.service.DebtPositionService;
+import it.gov.pagopa.pu.debtpositions.service.update.DebtPositionCancelInstallmentServiceImpl;
+import it.gov.pagopa.pu.debtpositions.util.InstallmentUtils;
+import it.gov.pagopa.pu.organization.dto.generated.Organization;
+import it.gov.pagopa.pu.organization.dto.generated.OrganizationStatus;
+import jakarta.transaction.Transactional;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+
+import java.util.Collection;
+import java.util.List;
+
+@Service
+@Slf4j
+public class DebtPositionDeletionServiceImpl implements DebtPositionDeletionService {
+
+  private final DebtPositionService debtPositionService;
+  private final OrganizationService organizationService;
+  private final AuthorizeOperatorOnDebtPositionTypeService authorizeOperatorOnDebtPositionTypeService;
+  private final DebtPositionCancelInstallmentServiceImpl debtPositionCancelInstallmentService;
+
+  public DebtPositionDeletionServiceImpl(DebtPositionService debtPositionService, OrganizationService organizationService, AuthorizeOperatorOnDebtPositionTypeService authorizeOperatorOnDebtPositionTypeService, DebtPositionCancelInstallmentServiceImpl debtPositionCancelInstallmentService) {
+    this.debtPositionService = debtPositionService;
+    this.organizationService = organizationService;
+    this.authorizeOperatorOnDebtPositionTypeService = authorizeOperatorOnDebtPositionTypeService;
+    this.debtPositionCancelInstallmentService = debtPositionCancelInstallmentService;
+  }
+
+  @Override
+  @Transactional
+  public String deleteDebtPosition(Long debtPositionId, String accessToken, String operatorExternalUserId) {
+    log.info("Cancelling debt position having id {}", debtPositionId);
+
+    DebtPosition debtPosition = debtPositionService.getDebtPositionNoPII(debtPositionId);
+
+    if (DebtPositionStatus.DRAFT.equals(debtPosition.getStatus())) {
+      deleteDraftDebtPosition(debtPosition, accessToken, operatorExternalUserId);
+      return null;
+    }
+
+    if(isIunPresent(debtPosition)) {
+      throw new ConflictErrorException("The debt position with id " + debtPositionId + " cannot be deleted because it is been notified");
+    }
+
+    if(!InstallmentUtils.DELETABLE_DP_STATUSES.contains(debtPosition.getStatus())){
+      throw new ConflictErrorException("The debt position with id " + debtPositionId + " cannot be deleted because is not in allowed status: " + debtPosition.getStatus());
+    }
+
+    DebtPositionDTO debtPositionDTO = debtPositionService.mapDebtPosition(debtPosition);
+
+    List<InstallmentDTO> installment2operate = debtPositionDTO.getPaymentOptions().stream()
+      .map(PaymentOptionDTO::getInstallments).flatMap(Collection::stream)
+      .filter(installmentDTO -> !InstallmentStatus.CANCELLED.equals(installmentDTO.getStatus()))
```

Comment:
you should skip also INVALID
```suggestion
      .filter(installmentDTO -> InstallmentUtils.MODIFIABLE_STATUSES.contains(installmentDTO.getStatus()))
```